docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,408 @@ title: Data Grid - Editing
+Once the row was saved, `processRowUpdate` must return the row object that will be used to update the internal state.
+This value returned is used as argument to a call to `apiRef.current.updateRows`.
 
-You can override the default [start editing](#start-editing) triggers using the [`event.defaultMuiPrevented`](/components/data-grid/events#disabling-the-default-behavior) on the synthetic React events.
+If while calling `processRowUpdate` you need to cancel the save process, like when a database validation fails or the user wants to reject the changes, there are two options:
 
-{{"demo": "StartEditButtonGrid.js", "bg": "inline", "disableAd": true}}
+- Reject the promise to not update the internal state and keep the cell in edit mode.
+- Resolve the promise with the second argument (the original row) received to make it stop silently, not updating the internal state but exiting the edit mode.
 
-### Events [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+The next demo implements the first option.
+Instead of [validating](#validation) while typing, it validates in the server.
+If the new name is empty it will not allow to save and the cell is kept into edit mode.
+Also, it shows that `processRowUpdate` can be used to pre-process the row object that will be saved into the internal state.
+To leave the edit mode, press <kbd class="key">Escape</kbd> or enter a valid name.
 
-The editing feature leverages the event capability of the grid and the apiRef.
-The following events can be imported and used to customize the edition:
+{{"demo": "ServerSidePersistence.js", "bg": "inline", "defaultCodeOpen": false}}
 
-- `cellEditStart`: emitted when the cell turns to edit mode.
-- `cellEditStop`: emitted when the cell turns back to view mode.
-- `cellEditCommit`: emitted when the new value is committed.
-- `editCellPropsChange`: emitted when the props passed to the edit cell component are changed.
+### Ask confirmation before save
 
-Catching events can be used to add a callback after an event while ignoring its triggers.
+The second option to cancel the save process allows the user to reject the changes and return immediatelly to the view mode.
+This is acomplished by resolving `processRowUpdate` with the second argument.
 
-The demo shows how to catch the start & end edit events to log which cell is editing in an info message:
+The following demo shows how this approach can be leveraged to ask a confirmation before sending the data to the server.
+In case the user accepts the change, the internal state is updated with the values.
+However, if the changes are rejected, no state is changed and the cell is reverted to its original value.
+The demo also employs validation to prevent entering an empty name.
 
-{{"demo": "CatchEditingEventsGrid.js", "bg": "inline", "disableAd": true}}
+{{"demo": "AskConfirmationBeforeSave.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
Other columns (date created and last login) are never saved. They should be removed or not editable

Reply:
True, I disabled them.